modules/brew/brew.sh

```diff
@@ -0,0 +1,177 @@
+
+# Always install Brew
+echo "Downloading and installing Brew..."
+curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh > /tmp/brew.sh
+chmod +x /tmp/brew.sh
+/tmp/brew.sh
+
+# Move Brew installation and set ownership to default user (UID 1000)
+tar --zstd -cvf /usr/share/homebrew.tar.zst /home/linuxbrew/.linuxbrew
+cp -R /home/linuxbrew /usr/share/homebrew
+chown -R 1000:1000 /usr/share/homebrew
+
+# Write systemd service files dynamically
+cat >/usr/lib/systemd/system/brew-setup.service <<EOF
+[Unit]
+Description=Setup Brew
+Wants=network-online.target
+After=network-online.target
+ConditionPathExists=!/etc/.linuxbrew
+ConditionPathExists=!/var/home/linuxbrew/.linuxbrew
+
+[Service]
+Type=oneshot
+ExecStart=/usr/bin/mkdir -p /tmp/homebrew
+ExecStart=/usr/bin/tar --zstd -xvf /usr/share/homebrew.tar.zst -C /tmp/homebrew
+ExecStart=/usr/bin/cp -R -n /tmp/homebrew/home/linuxbrew/.linuxbrew /var/home/linuxbrew
+ExecStart=/usr/bin/chown -R 1000:1000 /var/home/linuxbrew
+ExecStart=/usr/bin/rm -rf /tmp/homebrew
+ExecStart=/usr/bin/touch /etc/.linuxbrew
+
+[Install]
+WantedBy=default.target multi-user.target
+EOF
+
+cat >/usr/lib/systemd/system/brew-update.service <<EOF
+[Unit]
+Description=Auto update brew for mutable brew installs
+After=local-fs.target
+After=network-online.target
+ConditionPathIsSymbolicLink=/home/linuxbrew/.linuxbrew/bin/brew
+
+[Service]
+User=1000
+Type=oneshot
+Environment=HOMEBREW_CELLAR=/home/linuxbrew/.linuxbrew/Cellar
+Environment=HOMEBREW_PREFIX=/home/linuxbrew/.linuxbrew
+Environment=HOMEBREW_REPOSITORY=/home/linuxbrew/.linuxbrew/Homebrew
+ExecStart=/usr/bin/bash -c "/home/linuxbrew/.linuxbrew/bin/brew update"
+EOF
+
+cat >/usr/lib/systemd/system/brew-upgrade.service <<EOF
+[Unit]
+Description=Upgrade Brew packages
+After=local-fs.target
+After=network-online.target
+ConditionPathIsSymbolicLink=/home/linuxbrew/.linuxbrew/bin/brew
+
+[Service]
+User=1000
+Type=oneshot
+Environment=HOMEBREW_CELLAR=/home/linuxbrew/.linuxbrew/Cellar
+Environment=HOMEBREW_PREFIX=/home/linuxbrew/.linuxbrew
+Environment=HOMEBREW_REPOSITORY=/home/linuxbrew/.linuxbrew/Homebrew
+ExecStart=/usr/bin/bash -c "/home/linuxbrew/.linuxbrew/bin/brew upgrade"
+EOF
+
+# Write systemd timer files dynamically
+cat >/usr/lib/systemd/system/brew-update.timer <<EOF
+[Unit]
+Description=Timer for brew update for mutable brew
+Wants=network-online.target
+
+[Timer]
+OnBootSec=${WAIT_AFTER_BOOT_UPDATE}
+OnUnitInactiveSec=${UPDATE_INTERVAL}
+Persistent=true
+
+[Install]
+WantedBy=timers.target
+EOF
+
+cat >/usr/lib/systemd/system/brew-upgrade.timer <<EOF
+[Unit]
+Description=Timer for brew upgrade for on image brew
+Wants=network-online.target
+
+[Timer]
+OnBootSec=${WAIT_AFTER_BOOT_UPGRADE}
+OnUnitInactiveSec=${UPGRADE_INTERVAL}
+Persistent=true
+
+[Install]
+WantedBy=timers.target
+EOF
+
+# Copy shell configuration files
+cp -r "${MODULE_DIRECTORY}"/brew/brew-fish-completions.fish /usr/share/fish/vendor_conf.d/brew-fish-completions.fish
```

Comment:
Nitpick, but please rename this line & the file to `brew-fish-completion.fish` instead of `brew-fish-completions.fish`

Bash completion is named `brew-bash-completion.sh`, so it makes sense to rename it appropriately.

Reply:
Isn't `conversations` the proper form? I feel like I've seen that more often.

Reply:
@xynydev

`completions` might make more sense when I think about it, since it indicates various command completions for `brew`. Command completion indicates that brew can only execute 1 command which gets the completion, which is not true.

Easy change, I'll do it.

Reply:
Did it